frontend/src/Components/EventPreviewCard.jsx

```diff
@@ -49,12 +51,26 @@ export default function EventPreviewCard(props) {
             reload,
             setReload,
         });
-        props.setTargetEvent(eventDataObj);
+
+        if (!barrier) {
+            props.setTargetEvent(eventDataObj);
+        }
     }
     useEffect(setEventAsTarget, [loggedUser, volunteersList, loggedVolunteerPartOfEvent, loggedVolunteerRequestAccepted, volunteersEmailList]);
 
+    
     const handleClickOnEvent = () => {
-        setEventAsTarget();
+        const eventDataObj = Object.assign({}, event, {
+            volunteersList,
+            loggedVolunteerPartOfEvent,
+            loggedVolunteerRequestAccepted,
+            volunteersEmailList,
+            reload,
+            setReload,
+        });
+        props.setTargetEvent(eventDataObj);
+        setBarrie(false);
```

Comment:
here and on line 13 above, what's a "barrie?"

Reply:
it's a variable created to prevent the first useEffect, this is what we talked about the night before, it's what you and Briahana suggested to avoid `component did mount`

Reply:
just noticed the spelling mistake, thank you :)